### PR TITLE
Added protection when submitting new user settings.

### PR DESCRIPTION
**Pull request by xIvan0ff**
Updated the code for the `submitInfo()` function in the **UCP** module.

### CODE REVIEW

File: application/modules/ucp/controllers/settings.php [hunk 1]
```diff
@@ -90,9 +90,18 @@ public function submitInfo()
 		$this->load->model("settings_model");
 
 		// Gather the values
+
+		$nickname = $this->input->post("nickname");
+		$location = $this->input->post("location");
+
+		if(!(is_string($nickname)) || !(is_string($location)))
```
Comment (alexandre433) [inline application/modules/ucp/controllers/settings.php:97]: too many parentheses, `!is_string($variable)` is enough

File: application/modules/ucp/controllers/settings.php [hunk 2]
```diff
@@ -90,9 +90,18 @@ public function submitInfo()
...
+		{
+			die("4");
+		}
+
 		$values = array(
-			'nickname' => htmlspecialchars($this->input->post("nickname")),
-			'location' => htmlspecialchars($this->input->post("location")),
+			'nickname' => $this->template->format($nickname, false, false, true, false),
```
Comment (alexandre433) [inline application/modules/ucp/controllers/settings.php:103]: Could you add a comment, before this part, stating what this does?

**Comment by disclosurez** [comment]
some sort of vulnerability fix?

**Comment by alexandre433** [comment]
for now, we're just sanding the edges

**Comment by lastdarken** [comment]
We Need to Perform a security "Audit" PHP application?

**Comment by lastdarken** [comment]
@xIvan0ff
please check:

https://github.com/FusionGen/FusionGen/issues/10

https://github.com/FusionGen/FusionGen/issues/18

**Comment by lastdarken** [comment]
@xIvan0ff

Please check:

#10

#18

**Comment by omghixd** [comment]
These injections happened before the last commit. Thus those incidences cannot apply to new commits.

**Pull request merged**